### PR TITLE
mongodb: fix findOne<T> options to be optional again

### PR DESCRIPTION
**Pull request by stern-shawn**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [n/a] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [n/a] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

Previous PR #51003  broke the optional definition of `options`, so that if users are trying to use `findOne` with a generic and no options, they get type errors even if they were previously doing so with no type or runtime errors.

image

Fixing the definition of `options` back to optional for `findOne` with a defined generic.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## mongodb ([<kbd>unpkg</kbd>](https://unpkg.com/browse/mongodb@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'mongodb'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7db96288b9493cebf06ed313622866d4b9b4f101
</p>

**Comment by stern-shawn** [comment]
@LinusU @emmanuelgautier Could either of you take a look, please? Thank you!

File: types/mongodb/index.d.ts [hunk 1]
```diff
@@ -1371,7 +1371,7 @@ export interface Collection<TSchema extends { [key: string]: any } = DefaultSche
     ): Promise<TSchema | null>;
     findOne<T = TSchema>(
         filter: FilterQuery<TSchema>,
-        options: FindOneOptions<T extends TSchema ? TSchema : T>,
+        options?: FindOneOptions<T extends TSchema ? TSchema : T>,
```
Comment (emmanuelgautier) [inline types/mongodb/index.d.ts:1374]: You should add a test calling this function without the `options` parameter to prevent future regression

**Review by emmanuelgautier — CHANGES_REQUESTED** [review]

File: types/mongodb/index.d.ts (hunk 1 above)
Comment (stern-shawn) [reply types/mongodb/index.d.ts:1374]: Right, that's why I made this small edit to the existing test [here](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/51073/files#diff-6b5da2618493adf61d45da47ff1a6f6721cea888c0d7e94a5e8754988ad6d342R84)

There is then coverage for:
- `findOne` with no generic and no options
- `findOne` with a generic and no options
- `findOne` with a generic and options

What did I miss?

Comment (emmanuelgautier) [reply types/mongodb/index.d.ts:1374]: My bad

**Review by emmanuelgautier — APPROVED** [review]

**Comment by emmanuelgautier** [comment]
Ready to merge

File: types/mongodb/index.d.ts (hunk 1 above)
Comment (stern-shawn) [reply types/mongodb/index.d.ts:1374]: All good, thanks for making sure I did this right!

**Comment by LinusU** [comment]
@stern-shawn @emmanuelgautier sorry for not getting to this earlier, I was offline last week 😅 

I'm not sure that I think that this the correct approach forward. Would you mind sharing some more details on what your code looked like?

It seems like you are trying to change the return value from the collection, but without providing a `projection` to actually augment the schema type 🤔 

As listed under ["Common Misstakes"](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes) a type parameter should _not_ be used to coerce the return value (the `getMeAT` point), which seems to maybe be what you are doing in your code with this change? 🤔 

Would love some more information!

**Pull request merged**